File: Leetcode/cpp/1557.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<bool> incomingEdges(n, false);
        for (vector<int>& edge : edges) {
            incomingEdges[edge[1]] = true;
        }
        vector<int> output;
        for (int i = 0; i < n; i++) {
            if (!incomingEdges[i]) {
                output.push_back(i);
            }
        }
        return output;
    }
};

File: Leetcode/Python/_1557.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        output = set(list(range(n)))
        for u, v in edges:
            if v in output:
                output.remove(v)
                
        return output
    
    
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        indegree = [0] * n
        for i, j in edges:
            indegree[j] += 1
        lst = []
        for i in range(n):
            if indegree[i] == 0:
                lst.append(i)
        return lst
    
